src/lib/builders/scroll-area/create.ts

```diff
@@ -0,0 +1,524 @@
+	scrollbarXEl: WithGet<Writable<HTMLElement | null>>;
+	scrollbarYEl: WithGet<Writable<HTMLElement | null>>;
+	scrollAreaEl: WithGet<Writable<HTMLElement | null>>;
+	viewportEl: WithGet<Writable<HTMLElement | null>>;
+	contentEl: WithGet<Writable<HTMLElement | null>>;
+	options: ToWritableStores<Required<Omit<CreateScrollAreaProps, 'ids'>>>;
+	ids: ToWritableStores<IdObj<ScrollAreaIdParts>>;
+};
+
+export type ScrollAreaScrollbarState = {
+	isHorizontal: WithGet<Writable<boolean>>;
+	domRect: WithGet<Writable<DOMRect | null>>;
+	prevWebkitUserSelect: WithGet<Writable<string>>;
+	pointerOffset: WithGet<Writable<number>>;
+	thumbEl: WithGet<Writable<HTMLElement | null>>;
+	thumbOffset: WithGet<Writable<number>>;
+	scrollbarEl: WithGet<Writable<HTMLElement | null>>;
+	sizes: WithGet<Writable<Sizes>>;
+	orientation: WithGet<Writable<Orientation>>;
+	hasThumb: WithGet<Readable<boolean>>;
+	isVisible: WithGet<Writable<boolean>>;
+	handleWheelScroll: (e: WheelEvent, payload: number) => void;
+	handleThumbDown: (payload: { x: number; y: number }) => void;
+	handleThumbUp: (e: MouseEvent) => void;
+	onThumbPositionChange: () => void;
+	onDragScroll: (payload: number) => void;
+	handleSizeChange: () => void;
+};
+
+export type ScrollAreaState = {
+	rootState: ScrollAreaRootState;
+	scrollbarState: ScrollAreaScrollbarState;
+};
+
+const defaults = {
+	type: 'hover' as const,
+	hideDelay: 600,
+	dir: 'ltr' as const,
+};
+
+export function createScrollArea(props?: CreateScrollAreaProps) {
+	const withDefaults = { ...defaults, ...props };
+
+	const options = toWritableStores(omit(withDefaults, 'ids'));
+
+	const cornerWidth = withGet.writable(0);
+	const cornerHeight = withGet.writable(0);
+	const scrollbarXEnabled = withGet.writable(false);
+	const scrollbarYEnabled = withGet.writable(false);
+
+	const scrollAreaEl = withGet.writable<HTMLElement | null>(null);
+	const viewportEl = withGet.writable<HTMLElement | null>(null);
+	const contentEl = withGet.writable<HTMLElement | null>(null);
+	const scrollbarXEl = withGet.writable<HTMLElement | null>(null);
+	const scrollbarYEl = withGet.writable<HTMLElement | null>(null);
+
+	const ids = toWritableStores({ ...generateIds(scrollAreaIdParts), ...withDefaults.ids });
+
+	const rootState: ScrollAreaRootState = {
+		cornerWidth,
+		cornerHeight,
+		scrollbarXEnabled,
+		scrollbarYEnabled,
+		viewportEl,
+		contentEl,
+		options,
+		scrollbarXEl,
+		scrollbarYEl,
+		scrollAreaEl,
+		ids,
+	};
+
+	const root = makeElement(name(), {
+		stores: [cornerWidth, cornerHeight, ids.root],
+		returned: ([$cornerWidth, $cornderHeight, $rootId]) => {
+			return {
+				style: styleToString({
+					position: 'relative',
+					'--melt-scroll-area-corner-width': `${$cornerWidth}px`,
+					'--melt-scroll-area-corner-height': `${$cornderHeight}px`,
+				}),
+				id: $rootId,
+			};
+		},
+		action: (node: HTMLElement) => {
+			scrollAreaEl.set(node);
+			return {
+				destroy() {
+					scrollAreaEl.set(null);
+				},
```

Comment:
Maybe an update checking if the currEl is the root being destroyed would be better

Reply:
What do you mean by this? The `scrollAreaEl` is scoped to this specific instance so if this one is destroyed it would be the root right?